mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala

```diff
@@ -325,8 +350,13 @@ final class OneVsRest @Since("1.4.0") (
       multiclassLabeled.persist(StorageLevel.MEMORY_AND_DISK)
     }
 
+    val iters = Range(0, numClasses).par
+    iters.tasksupport = new ForkJoinTaskSupport(
+      new ForkJoinPool(Math.min(getParallelism, numClasses))
```

Comment:
I don't think `ForkJoinPool` is the best thing to use here.  It's more geared for a bunch of small tasks that might spawn other tasks.  Just a regular thread-pool is fine.  Also, it is a little better for the case of `parallelism` 1 to use `sameThreadExecutor`.  This will just run in the current thread and has no overhead (equivalent to running in serial), where a thread-pool of size 1 will still create another thread and does have some overhead.